openlr_dereferencer/maps/abstract.py

```diff
@@ -1,11 +1,57 @@
-"""An abstract `MapReader` base class, which must be implemented for each
-map format to decode location references on."""
+"""Contains an abstract `MapReader` base class, which must be implemented for each
+map format to decode location references on.
+
+A MapReader is an interface with which the decoder can traverse the map. An implementation may
+consist of a database connection or something similar. Its purpose is to let the decoder read
+map objects.
+
+In order to implement a reader for a new map, the following interfaces have to be fulfilled:
+
+* :py:attr:`~MapReader`
+* :py:attr:`~Node`
+* :py:attr:`~Line`
+
+The map model
+-------------
+Nodes
+=====
+A node is an object with an ID and a WGS84 longitude/latitude position.
+
+Lines
+=====
+A line interconnects exactly two nodes in exactly one direction. As additional attributes,
+it has a shape, a functional road class and a form of way.
+
+Note that the map you want to implement may have a different line concept. For example, it is
+common that roads are not necessarily directed. Also, roads may connect more than two nodes.
+
+Let's consider an example map format that defines ways like this:
+
+.. image:: _static/Mapformat_1_Way.svg
+
+So, multi-node, multi-directed ways.
+
+If we write a MapReader adapter for this format, we could consider every
+directed link between points being a line, yielding multiple lines per way.
```

Comment:
I would use the word `nodes` rather than `points` here to be consistent with the previous lines. This would make it `every directed link between nodes`.